src/Sentry.Profiling/SampleProfilerSession.cs

```diff
@@ -0,0 +1,46 @@
+using System.Diagnostics.Tracing;
+using Microsoft.Diagnostics.NETCore.Client;
+using Microsoft.Diagnostics.Tracing.EventPipe;
+using Microsoft.Diagnostics.Tracing.Parsers;
+
+namespace Sentry.Profiling;
+
+internal class SampleProfilerSession
+{
+    private DiagnosticsClient _client;
+    private EventPipeSession _session;
+    private MemoryStream _stream = new MemoryStream();
```

Comment:
Specifically, consider long-running transactions.  The `EventPipeSession` uses a circular buffer, but we're copying continuously from the buffer to this memory stream.  How big will it grow?  What if there's just one transaction for the lifetime of the application?  Seems like we need to impose some limitations?

Reply:
We're limiting the profile collection to 30 seconds (from https://develop.sentry.dev/sdk/profiles/#validation) in `SamplingTransactionProfiler` - creating a delayed task that stops the profiler:
```c#
        Task.Delay(timeoutMs, _cancellationToken).ContinueWith(_ =>
        {
            if (Stop(TimeSpan.FromMilliseconds(timeoutMs)))
            {
                _options.LogDebug("Profiling is being cut-of after {0} ms because the transaction takes longer than that.", timeoutMs);
            }
        }, CancellationToken.None);
```

Reply:
Ok.  That makes sense.  And I also see where the stream is ultimately disposed (in `CreateTraceLogAsync`).  But where is the `EventPipeSession` disposed?

Reply:
I've added an explicit `Dispose()` to `FinishAsync()`